### PR TITLE
17 change data visualization using matplotlib

### PR DESCRIPTION
**Pull request by ldxFAIRYTAIL**
Fixed：#17
change data visualization using matplotlib

### CODE REVIEW

File: strix/data_checker.py [hunk 1]
```diff
@@ -238,23 +220,26 @@ def get_train_valid_datasets():
             for i, data in enumerate(tqdm(dataloader)):
                 bs = dataloader.batch_size
                 if exist_mask and overlap:
-                    mask_class_num = len(data[msk_key].unique())
-                    if mask_class_num > 2:
-                        msk = one_hot(data[msk_key], mask_class_num, dim=1).type(torch.bool)
+                    mask_class_num = len(data[msk_key].unique()) - 1
```
Comment (ChenglongWang) [inline strix/data_checker.py]: what if `mask_class_num` = 1, program will crash? Maybe it's better to set `msk = None` if `mask_class_num` = 1. which mean only background.

File: strix/utilities/utils.py [hunk 2]
```diff
@@ -520,145 +523,40 @@ def is_bound(data, coord, offsets):
...
+    if color_num >0:
+        colors = get_colors(color_num)
+        cmap = get_colormaps(color_num)
+    for i in range(nrow):
+        for j in range(ncol):
+            axes[i,j].axis('off')
+            if i* ncol+j < images.shape[0]:
+                axes[i,j].imshow(images[i*ncol+j,...].squeeze(), cmap = 'gray', interpolation = 'bicubic')
+                if method == 'mask':
+                    axes[i,j].imshow(np.ma.masked_equal(masks[i*ncol+j,...].squeeze(), 0), cmap, alpha = alpha, norm=matplotlib.colors.Normalize(vmin=1, vmax=color_num))
```
Comment (ChenglongWang) [inline strix/utilities/utils.py]: `matplotlib.colors.Normalize()` -> `mcolors.Normalize()` or just 
`from matplotlib.colors import LinearSegmentedColormap, Normalize`

`alpha = alpha` -> `alpha=alpha`, no spacing.

File: strix/utilities/utils.py [hunk 3]
```diff
@@ -520,145 +523,40 @@ def is_bound(data, coord, offsets):
...
+                elif method == 'contour':
+                    list = [i for i in np.unique(masks[i*ncol+j,...].squeeze()).tolist() if i != 0 and i is not None]
```
Comment (ChenglongWang) [inline strix/utilities/utils.py]: equivalent to `[i for i in np.unique(masks[i*ncol+j,...].squeeze()).tolist() if i]`

File: strix/utilities/utils.py [hunk 4]
```diff
@@ -520,145 +523,40 @@ def is_bound(data, coord, offsets):
...
+                    if len(list) > 0:
+                        axes[i,j].contour(masks[i*ncol+j,...].squeeze(), levels = [x-0.01 for x in list], colors = colors[min(list)-1:max(list)])
```
Comment (ChenglongWang) [inline strix/utilities/utils.py]: no spacing around `=`

**Review by ChenglongWang — CHANGES_REQUESTED** [review]

File: strix/data_checker.py (hunk 1 above)
Comment (ldxFAIRYTAIL) [reply strix/data_checker.py]: I have added a function to check `mask_class_num` whether the mask is single-class or multi-class.

**Review by ChenglongWang — APPROVED** [review]

**Pull request merged**